### PR TITLE
User registration

### PR DESCRIPTION
**Pull request by vivekshah1801**

### CODE REVIEW

**Comment by vivekshah1801** [comment]
@YashKumarVerma waiting for your review.

File: src/modules/user/models.ts [hunk 1]
```diff
@@ -34,6 +36,39 @@ class UserModel {
...
+   * @returns UserDetails
+   */
+  static async registerUser(user: UserDetails): Promise<UserDetails> {
+    return new Promise((resolve, reject) => {
+      /** if something goes wrong, send data with reject({ .... }) */
+      /**   if everything is alright, send data with resolve */
+      
+      if( !user || !user.username || !user.firstname || !user.password )
+      {
+        reject("User Details Invalid")
```
Comment (YashKumarVerma) [inline src/modules/user/models.ts:54]: return always json response.

can use something like 
```
{
erorr: true/false,
message: "why the error is true"},
payload: {actual data here}
}
```

**Comment by YashKumarVerma** [comment]
update travis.yml to fix build failing.

**Comment by vivekshah1801** [comment]
closing, due to #37

**Pull request closed without merging**